### ISSUE TITLE
About background colors?

### ISSUE THREAD

**Issue by tikboaHIT** [kwea123/ngp_pl#32]
Among them, [troch-ngp](https://github.com/ashawkey/torch-ngp/blob/87bb685b3f019d3cc9ddda49d5e569e0209b00a2/nerf/utils.py#L428-L444) uses a random background color
[nerf ](https://github.com/yenchenlin/nerf-pytorch/blob/master/run_nerf.py#L588-L591)uses a fixed background.
[ngp_pl](https://github.com/kwea123/ngp_pl/blob/master/datasets/nsvf.py#L99-L100) seems to use a fixed background.

> Q: Why are background colors randomized during NeRF training?
> A: Transparency in the training data indicates a desire for transparency in the learned model. Using a solid background color, the model can minimize its loss by simply predicting that background color, rather than transparency (zero density). By randomizing the background colors, the model is forced to learn zero density to let the randomized colors "shine through".

As mentioned in instant-ngp, would it be better to use a random background?

**Comment by kwea123** [comment]
This is a valid point, I also observe that in Chair scene, a white part of the chair is predicted to be background, which causes drop in accuracy. I will try random bg color in training, and fixed color in testing.

**Comment by kwea123** [comment]
However, this only works when alpha channel is known, therefore only a little part of data (some Synthetic-NSVF data don't have alpha channel...) so I don't know if it's worth implementing at the moment. I'd like to work on more substantial part first. Let me know if you have some findings!

**Comment by kwea123** [comment]
added for real data training and it improves quality. As for synthetic data, it would require setting the same random color for the data and the network, which is not straightforward, so I'll leave it for the future...
